lib/spack/spack/cmd/install.py

```diff
@@ -50,7 +50,7 @@ def setup_parser(subparser):
         '--dont-restage', action='store_true',
         help="if a partial install is detected, don't delete prior state")
     subparser.add_argument(
-        '--use-cache', action='store_true', dest='use_cache',
+        '--no-cache', action='store_false', dest='use_cache',
```

Comment:
@becker33  this should still probably be `store_true` (in which case by default, it's false, which means that by default you use the cache). `dest` should be updated too